src/facilities-context/__tests__/FacilitiesContext.test.tsx

```diff
@@ -0,0 +1,194 @@
+jest.mock("../../scenario-context/useScenario");
+jest.mock("../../database/index");
+jest.mock("../../feature-flags");
+
+import { renderHook } from "@testing-library/react-hooks";
+import JestMockPromise from "jest-mock-promise";
+import { cloneDeep, isEmpty, noop } from "lodash";
+import React from "react";
+import { mocked } from "ts-jest/utils";
+
+import {
+  compositeFacility,
+  referenceFacility,
+  userFacility,
+} from "../__fixtures__";
+import {
+  getFacilities,
+  getReferenceFacilities,
+  referenceFacilitiesProp,
+} from "../../database/index";
+import { useFlag } from "../../feature-flags";
+import { Scenario } from "../../page-multi-facility/types";
+import useScenario from "../../scenario-context/useScenario";
+import { FacilitiesProvider, useFacilities } from "../FacilitiesContext";
+
+const mockedUseScenario = mocked(useScenario);
+const mockedUseFlag = mocked(useFlag);
+const mockedGetReferenceFacilities = mocked(getReferenceFacilities);
+const mockedGetFacilities = mocked(getFacilities);
+
+function getTestHook() {
+  // setup unit under test
+  const wrapper: React.FC = ({ children }) => (
+    <FacilitiesProvider>{children}</FacilitiesProvider>
+  );
+  return renderHook(() => useFacilities(), { wrapper });
+}
+
+describe("FacilitiesContext", () => {
+  let mockBaselineScenario: Scenario;
+  beforeEach(() => {
+    // setup mocks
+    mockedUseFlag.mockReturnValue(true);
+    mockedUseScenario.mockReturnValue([
+      {
+        loading: true,
+        failed: false,
+        data: null,
+      },
+      noop,
+    ]);
+
+    mockBaselineScenario = {
+      name: "Mock Baseline",
+      baseline: true,
+      id: "mock-baseline-id",
+      [referenceFacilitiesProp]: { [userFacility.id]: referenceFacility.id },
+      useReferenceData: true,
+    } as Scenario; // we are missing some fields but they should not be relevant here
+  });
+
+  it("updates reference facilities when the scenario changes", async () => {
+    const { result, rerender, waitForNextUpdate } = getTestHook();
+    expect(result.current.state.referenceFacilities).toEqual({});
+
+    mockedUseScenario.mockReturnValue([
+      {
+        loading: false,
+        failed: false,
+        data: mockBaselineScenario,
+      },
+      noop,
+    ]);
+    mockedGetFacilities.mockResolvedValue([cloneDeep(userFacility)]);
+    mockedGetReferenceFacilities.mockResolvedValue([
+      cloneDeep(referenceFacility),
+    ]);
+
+    rerender();
+
+    await waitForNextUpdate();
+
+    expect(result.current.state.referenceFacilities).toEqual({
+      [referenceFacility.id]: referenceFacility,
+    });
+  });
+
+  it("does not use reference data when feature is toggled off", async () => {
```

Comment:
love these tests! I was wondering if you think it's enough here to check that with `useReferenceData=false` we don't see a composite facility, or if we would want to check that `state.referenceFacilities` also does not populate?

Reply:
no because `referenceFacilities` can still populate if the feature is toggled off ... we need it for stuff like figuring out if the scenario is even eligible to have the feature turned on (e.g. data quality checks - there must be a certain number of facilities with recent reference data or we don't let you turn it on)